### PR TITLE
Fix issue with aks-create Taskfile target failing due to no specified DOCKER_REGISTRY

### PR DESCRIPTION
**Pull request by matthchr**
Also fix unrelated warning from Dockerfiles for using FROM and as with different cases.

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains tests
- [ ] this PR contains YAML Samples

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -1226,6 +1229,14 @@ tasks:
...
       - 'if [ -z "{{.DOCKER_REGISTRY}}" ]; then echo "Error: DOCKER_REGISTRY must be set"; exit 1; fi'
       - az acr login --name {{.DOCKER_REGISTRY}}
+    vars:
+      DOCKER_REGISTRY:
+        sh: |
+          if [ ! -z "${DOCKER_REGISTRY}" ]; then
+            echo "${DOCKER_REGISTRY}"
+          else
+            echo "{{.DOCKER_REGISTRY}}"
+          fi
```
Comment (theunrepentantgeek) [inline Taskfile.yml:1239]: I don't think you need to do that. Task [will look in the environment](https://taskfile.dev/usage/#variables) for `.DOCKER_REGISTRY` if it's not already in scope.

**Review by theunrepentantgeek — APPROVED** [review]
Overall looks good. I think a couple of things can be simplified though.

File: Taskfile.yml [hunk 2]
```diff
@@ -891,11 +893,12 @@ tasks:
           DIR: "{{.AKS_WORKLOAD_IDENTITY_PATH}}"
           RESOURCE_GROUP: "{{.HOSTNAME}}-aso-rg"
       - task: controller:install-cert-manager
-      - "az acr login --name {{.ACR_NAME}}"
+      # - "az acr login --name {{.ACR_NAME}}"
```
[the remark this answers is not in the document]
Comment (matthchr) [reply Taskfile.yml:896]: Deleted actually, fixed.

This is now done as part of the docker-push-multiarch

File: Taskfile.yml (hunk 1 above)
Comment (matthchr) [reply Taskfile.yml]: Tested without this and confirmed it works.

**Pull request merged**